File: University Projects/C# Projects/Design and Programming Environments/Proiect/model/SwimmingTest.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace model
{
    public class SwimmingTest : Entity<int>
    {
        public int Distance { get; }
        public string Style { get; }
        private List<Participant> Participants { get; }

        public SwimmingTest(int id, int distance, string style) : base(id)
        {
            Distance = distance;
            Style = style;
            Participants = new List<Participant>();
        }

        public void AddParticipant(Participant participant)
        {
            Participants.Add(participant);
        }
    }
}

File: University Projects/Java Projects/Design and Programming Environments/Proiect/Model/src/main/java/app/model/SwimmingTest.java
package app.model;

import jakarta.persistence.*;

import java.util.ArrayList;
import java.util.List;

@jakarta.persistence.Entity
@Table(name = "SwimmingTests")
public class SwimmingTest extends Entity<Integer>{
    private int distance;
    private String style;
    @Transient
    private List<Participant> participants;

    public SwimmingTest() { }

    public SwimmingTest(Integer id, int distance, String style) {
        super(id);
        this.distance = distance;
        this.style = style;
        this.participants = new ArrayList<>();
    }

    public int getDistance() {
        return distance;
    }

    public String getStyle() {
        return style;
    }

    public List<Participant> getParticipants() {
        return participants;
    }

    public void addParticipant(Participant participant) {
        participants.add(participant);
    }
}
